tests/components/home_connect/test_switch.py

```diff
@@ -0,0 +1,192 @@
+
+SETTINGS_STATUS = {
+    setting.pop("key"): setting
+    for setting in load_json_object_fixture("home_connect/settings.json")
+    .get("Washer")
+    .get("data")
+    .get("settings")
+}
+
+PROGRAM = "LaundryCare.Dryer.Program.Mix"
+
+
+@pytest.fixture
+def platforms() -> list[str]:
+    """Fixture to specify platforms to test."""
+    return [Platform.SWITCH]
+
+
+async def test_switches(
+    bypass_throttle: Generator[None, Any, None],
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    get_appliances: Mock,
+) -> None:
+    """Test switch entities."""
+    get_appliances.side_effect = get_all_appliances
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert await integration_setup()
+    assert config_entry.state == ConfigEntryState.LOADED
+
+
+@pytest.mark.parametrize(
+    ("entity_id", "status", "service", "state"),
+    [
+        (
+            "switch.washer_program_mix",
+            {BSH_ACTIVE_PROGRAM: {"value": PROGRAM}},
+            SERVICE_TURN_ON,
+            STATE_ON,
+        ),
+        (
+            "switch.washer_program_mix",
+            {BSH_ACTIVE_PROGRAM: {"value": ""}},
+            SERVICE_TURN_OFF,
+            STATE_OFF,
+        ),
+        (
+            "switch.washer_power",
+            {BSH_POWER_STATE: {"value": BSH_POWER_ON}},
+            SERVICE_TURN_ON,
+            STATE_ON,
+        ),
+        (
+            "switch.washer_power",
+            {BSH_POWER_STATE: {"value": BSH_POWER_OFF}},
+            SERVICE_TURN_OFF,
+            STATE_OFF,
+        ),
+        (
+            "switch.washer_power",
+            {
+                BSH_POWER_STATE: {"value": ""},
+                BSH_OPERATION_STATE: {
+                    "value": "BSH.Common.EnumType.OperationState.Inactive"
+                },
+            },
+            SERVICE_TURN_OFF,
+            STATE_OFF,
+        ),
+    ],
+)
+async def test_switch_functionality(
+    entity_id: str,
+    status: dict,
+    service: str,
+    state: str,
+    bypass_throttle: Generator[None, Any, None],
+    hass: HomeAssistant,
+    config_entry: MockConfigEntry,
+    integration_setup: Callable[[], Awaitable[bool]],
+    setup_credentials: None,
+    appliance: Mock,
+    get_appliances: MagicMock,
+) -> None:
+    """Test switch functionality."""
+    appliance.status.update(SETTINGS_STATUS)
+    appliance.get_programs_available.return_value = [PROGRAM]
+    get_appliances.return_value = [appliance]
+
+    assert config_entry.state == ConfigEntryState.NOT_LOADED
+    assert await integration_setup()
+    assert config_entry.state == ConfigEntryState.LOADED
+
+    appliance.status.update(status)
+    await hass.services.async_call(
+        DOMAIN, service, {"entity_id": entity_id}, blocking=True
+    )
+    assert hass.states.is_state(entity_id, state)
+
+
+@pytest.mark.parametrize(
```

Comment:
Why is the exception handling parameterized so thoroughly?

Reply:
Basically for coverage. The way the code being testing is setup is separate switch logic for programs and power.